### ISSUE TITLE
Tie up loose ends in partitioning story

### ISSUE THREAD

**Issue by Manangka** [Deltares/imod-python#658]
In GitLab by @luitjansl on Nov 21, 2023, 16:57

We have implemented partitioning models into smaller submodels that can be solved in parallel.

However there are several loose ends that we still need to take care of.

* The partitioning still needs to be fixed for the hfb package, the lake package and the uzf package
* We still need to compute the auxiliary variables cdist and anglex for unstructured grids
* Postprocessing step: the exchange flux should still be merged into the balance files
* Transport is still untested for partitioning, but parallel flow + transport is still under development on the Modflow side, so it's not urgent.

### Tasks
- [x] https://github.com/Deltares/imod-python/issues/659
- [ ] https://github.com/Deltares/imod-python/issues/660
- [ ] https://github.com/Deltares/imod-python/issues/661
- [x] https://github.com/Deltares/imod-python/issues/662
- [x] https://github.com/Deltares/imod-python/issues/663
- [x] https://github.com/Deltares/imod-python/issues/664
- [x] https://github.com/Deltares/imod-python/issues/548

**Comment by Manangka** [comment]
In GitLab by @JoerivanEngelen on Nov 22, 2023, 10:26

Relevant for the computation of aux variables for unstructured grids is this comment: https://gitlab.com/deltares/imod/imod-python/-/issues/479#note_1506589478

Here's the code how Modflow 6 computes geometric properties:
https://github.com/MODFLOW-USGS/modflow6/blob/develop/src/Model/ModelUtilities/DisvGeom.f90

Original code to compute geometric properties (where angledx is not computed entirely correctly)

```python
def set_geometric_properties(df: pd.DataFrame, grid: xu.Ugrid2d) -> None:
    """
    Compute the geometric properties for every connection: the length in the
    first cell (cl1), the length in the second cell (cl2), and the width of the
    connection (hwa).
    """
    face_centroids = grid.centroids
    edge_coordinates = grid.edge_node_coordinates[df["edge_index"]]
    centroid_i = face_centroids[df["i"]]
    centroid_j = face_centroids[df["j"]]
    
    U = np.diff(edge_coordinates, axis=1)[:, 0]
    Vi = centroid_i - edge_coordinates[:, 0]
    Vj = centroid_j - edge_coordinates[:, 0]
    length = np.linalg.norm(U, axis=1)
    anglex = np.atan2(U[:, 1], U[:, 0])
    anglex[anglex < 0] += 2 * np.pi
    df["cl1"] = np.abs(np.cross(U, Vi)) / length
    df["cl2"] = np.abs(np.cross(U, Vj)) / length
    df["hwva"] = length
    df["anglex"] = anglex
    return
```

**Comment by Manangka** [comment]
In GitLab by @JoerivanEngelen on Nov 22, 2023, 12:37

added #659 as child task

**Comment by Manangka** [comment]
In GitLab by @JoerivanEngelen on Nov 22, 2023, 12:37

added #660 as child task

**Comment by Manangka** [comment]
In GitLab by @JoerivanEngelen on Nov 22, 2023, 12:38

added #662 as child task

**Comment by Manangka** [comment]
In GitLab by @JoerivanEngelen on Nov 22, 2023, 12:39

added #663 as child task

**Comment by Manangka** [comment]
In GitLab by @JoerivanEngelen on Nov 22, 2023, 12:39

added #664 as child task

**Comment by Manangka** [comment]
In GitLab by @JoerivanEngelen on Nov 22, 2023, 12:45

marked this issue as related to #324

**Comment by Manangka** [comment]
In GitLab by @JoerivanEngelen on Nov 22, 2023, 12:45

marked this issue as related to #565

**Comment by Manangka** [comment]
In GitLab by @JoerivanEngelen on Nov 22, 2023, 12:50

added #548 as child task

**Comment by Manangka** [comment]
In GitLab by @JoerivanEngelen on Dec 7, 2023, 15:10

marked this issue as related to #683
